src/metax_api/tests/api/rest/base/views/files/write.py

```diff
@@ -268,6 +268,32 @@ def test_create_file_error_json_validation(self):
         self.assertEqual('Json path:' in response.data['file_characteristics'][0], True,
                          'The error should contain the json path')
 
+    def test_create_file_allowed_checksum_algorithm(self):
+        self.test_new_data['checksum']['algorithm'] = 'SHA-512'
+
+        response = self.client.post('/rest/files', self.test_new_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED, response.data)
+        self.assertEqual(response.data['checksum']['algorithm'], 'SHA-512')
+
+        self.test_new_data['identifier'] = 'urn:nbn:fi:csc-md5'
+        self.test_new_data['file_path'] = '/md5/filepath/md5-filename'
+        self.test_new_data['file_name'] = 'md5-filename'
+        self.test_new_data['checksum']['algorithm'] = 'MD5'
+
+        response = self.client.post('/rest/files', self.test_new_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED, response.data)
+        self.assertEqual(response.data['checksum']['algorithm'], 'MD5')
+
+    def test_create_file_not_allowed_checksum_algorithm(self):
+        self.test_new_data['checksum']['algorithm'] = "sha2"
```

Comment:
In addition to value ``sha2``, I recommend checking with values like ``sha-256`` or ``sha256`` which also should not pass.